src/main/java/com/austinv11/servicer/ServicerProcessor.java

```diff
@@ -115,4 +113,22 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
     public SourceVersion getSupportedSourceVersion() {
         return super.getSupportedSourceVersion();
     }
+
+    private static class Services {
+        private final Set<String> impls = new HashSet<>();
```

Comment:
You should probably wrap these with Collections.synchronizedSet()

Reply:
That would only matter if the Processor is called concurrently right?

Reply:
Yes, but I don't believe its defined that processors can't be concurrent. It probably doesn't matter for javac, but eclipse's compiler likes to be weird.

Reply:
Looking through dagger and auto, it looks like neither uses synchronized or concurrent collections? So it is probably fine. Also, the previous implementation was using hash sets just fine as the values of the hashmap.

Reply:
just in case there are problems, I synchronized it on the entire processor class. That way we don't have to deal with multiple threads reading&modifying the hashmap at the same time. Using a synchronized map wouldn't be enough in that case because while it ensures that the map maintains a valid state, it doesn't prevent reads/writes from other process calls being overleaved.